DQM/L1TMonitor/src/L1TStage2CaloLayer1.cc

```diff
@@ -83,16 +94,52 @@ void L1TStage2CaloLayer1::dqmAnalyze(const edm::Event& event,
         const uint32_t* amcPtr = amcData.dataPtr();
         FATevent = ((amcPtr[5] >> 16) & 0xf) == 5;
         additionalFB = (amcPtr[5] >> 15) & 0x1;
-        int lPhi = amcData.layer1Phi();
-        if (daqData.BXID() != amcData.BXID()) {
-          eventMonitors.bxidErrors_->Fill(lPhi);
+        if (((amcPtr[5] >> 14) & 0x1) == 0) {
+          int lPhi = amcData.layer1Phi();
+          if (daqData.BXID() != amcData.BXID()) {
+            eventMonitors.bxidErrors_->Fill(lPhi);
+          }
+          if (daqData.L1ID() != amcData.L1ID()) {
+            eventMonitors.l1idErrors_->Fill(lPhi);
+          }
+          // AMC payload header has 16 bit orbit number, AMC13 header is full 32
+          if ((daqData.orbitNumber() & 0xFFFF) != amcData.orbitNo()) {
+            eventMonitors.orbitErrors_->Fill(lPhi);
+          }
+        }
+        if (iFed == 1354 && daqData.nAMCs() == 7 && i == 3 && amcData.amcNo() == 7 && ((amcPtr[5] >> 14) & 0x1) == 1) {
+          card7flag1354 = true;
+          for (int j = 0; j < 6; j++) {
+            FED1354_slot7_bits[j] = amcPtr[j + 6] & 0xFFFFFFFF;
+          }
+        }
+        if (iFed == 1356 && daqData.nAMCs() == 7 && i == 3 && amcData.amcNo() == 7 && ((amcPtr[5] >> 14) & 0x1) == 1) {
+          card7flag1356 = true;
+          for (int j = 0; j < 6; j++) {
+            FED1356_slot7_bits[j] = amcPtr[j + 6] & 0xFFFFFFFF;
+          }
+        }
+        if (iFed == 1358 && daqData.nAMCs() == 7 && i == 3 && amcData.amcNo() == 7 && ((amcPtr[5] >> 14) & 0x1) == 1) {
+          card7flag1358 = true;
+          for (int j = 0; j < 6; j++) {
+            FED1358_slot7_bits[j] = amcPtr[j + 6] & 0xFFFFFFFF;
+          }
+        }
+      }
+    }
+  }
+
+  if (not caloLayer1OutOfRun) {
+    for (int iword = 0; iword < 6; iword++) {
+      for (int ibit = 0; ibit < 32; ibit++) {
+        if (card7flag1354 == 1 && ((FED1354_slot7_bits[iword] >> ibit) & 0x1) == 1) {
```

Comment:
```suggestion
        if (card7flag1354 && ((FED1354_slot7_bits[iword] >> ibit) & 0x1) == 1) {
```
`card7flag1354` is already a bool